pkgs/top-level/config.nix

```diff
@@ -36,6 +36,27 @@ let
       internal = true;
     };
 
+    /* System options */
+
+    localSystem = mkOption {
+      type = types.attrs;
+      apply = lib.systems.elaborate;
+      description = "Local system";
+    };
+
+    # FIXME! NOTE: Not all functions are congruent in Nix, @oxij really hopes this is just a bug
+    # > $ nix-instantiate --eval -E 'with import ./lib; let f = systems.elaborate; x = { system = "x86_64-linux"; }; in f x == f x'
+    # > false
```

Comment:
:scream:

Reply:
that's because `(f x).emulator` is LAMBDA, so it's a feature. 
```
nix-repl> f = x: x

nix-repl> f == f
false
```

Reply:
I see, then maybe I'll try to fix that one with some refactoring later.

Reply:
Currently work in progress:
- https://github.com/NixOS/nixpkgs/pull/238331